avm/res/aad/domain-service/tests/e2e/waf-aligned/main.test.bicep

```diff
@@ -0,0 +1,154 @@
+targetScope = 'subscription'
+
+metadata name = 'WAF-aligned'
+metadata description = 'This instance deploys the module in alignment with the best-practices of the Azure Well-Architected Framework.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-aad-domainservices-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'aaddswaf'
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableTelemetry bool = false
+
+@description('Optional. A token to inject into the name of each resource. This value can be automatically injected by the CI.')
+param namePrefix string = '#_namePrefix_#'
+
+var varDiagSettings = [
```

Comment:
I know why you used `vars` here, but please note - any element that is not part of the 

```bicep
module testDeployment '../../../main.bicep' = {
  scope: resourceGroup
  (...)
}
```
block will not find its way into the readme `usageExamples`.

Duplicatting the configuration in both the init & idem may seem, odd, yet I'd recommend to actually do that instead of using vars for the above reason.

Thoughts?

Reply:
That makes sense. Duplicating them again.